app-ng/src/job.lisp

```diff
@@ -0,0 +1,113 @@
+(in-package #:qvm-app-ng)
+
+(deftype job-status ()
+  '(member fraiche running finished interrupted error))
+
+(defstruct (job (:constructor %make-job))
```

Comment:
for a job, i'd like as many slots to be read-only as possible

Reply:
also specify the types to each slot

Reply:
This is difficult because the slots mostly are not provided to the constructor, and so mixing `nil` and `:type` is no bueno.